model_training/ml_trainmodel.py

```
@@ -19,46 +19,57 @@
 # create an empty dictionary that will eventually contain all of the loss metrics of an iteration
 loss_dict = {}
 
-def configure_loader(parms, dataset_train_list):
+def configure_loader(parms, dataset_train_list, dataset_test_list):
+    assert len(dataset_train_list) == len(dataset_test_list)
+    assert len(dataset_train_list) > 0
```

Updated ```configure_loader``` to accept both train and test dataset lists, and added checks (non-empty, matching lengths) so we fail fast if the dataset pairing is wrong.

---

I think the extra complexity of another function argument might not be worth it in this case. The loader should have a single clearly defined purpose: to load data. I think it sends the wrong message if one of the function arguments is data that it is not loading.